### PR TITLE
make job colour theme configurable

### PR DESCRIPTION
**Pull request by oliver-sanders**
First attempt at getting back into the UI.

Make the job theme configurable from the settings page.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [ ] Appropriate tests are included (unit and/or functional).
- [x] Appropriate change log entry included.
- [x] No documentation update required.

### CODE REVIEW

**Comment by oliver-sanders** [comment]
It works, however, there is a strange CSS quirk that I can't quite explain where changing the theme affects the display of the other themes in the selector:

theme

**Comment by oliver-sanders** [comment]
Deconflicted.

File: src/views/UserProfile.vue [hunk 1]
```diff
@@ -166,6 +232,35 @@ export default {
...
+    text-align: center;
+  }
+  .v-radio {
+    /* allow the element to center */
+    display: inline-block;
+    /* jobs are 1em wide, offset the element by half this */
+    padding-left: 0.5em;
+  }
+}
+</style>
```
Comment (oliver-sanders) [inline src/views/UserProfile.vue:266]: Happy to move into an `.scss` file, can't remember how to do it 😞.

File: src/views/UserProfile.vue [hunk 2]
```diff
@@ -139,18 +188,35 @@ import {
...
         decrease: mdiFormatFontSizeDecrease
-      }
+      },
+      jobStates: [
+        'submitted',
+        'running',
+        'succeeded',
+        'failed',
+        'submit-failed'
+      ],
```
Comment (oliver-sanders) [inline src/views/UserProfile.vue:213]: Should move to an enum somewhere.

File: src/views/UserProfile.vue [hunk 3]
```diff
@@ -121,6 +126,50 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
                 </v-btn>
               </v-flex>
             </v-layout>
+
+            <v-layout row align-center wrap>
+              <v-flex xs3>
+                <span>Colour Theme</span>
+              </v-flex>
+                <v-radio-group v-model="jobTheme" column>
+                  <table class="c-job-state-table">
```
Comment (oliver-sanders) [inline src/views/UserProfile.vue:135]: Messed around with Vuetify components, however, a good old table seems to do the best job.

**Comment by hjoliver** [comment]
Nice!

**Review by kinow — COMMENTED** [review]
Nice work!! :tada: 

I've created https://github.com/oliver-sanders/cylc-ui/pull/2 where I moved the value to Vuex. I didn't take much time thinking how to implement it.

So that should have what approach I would try first to get the theme to the `Job` component without altering attributes of DOM elements, and persisting after the page is refreshed.

File: src/views/UserProfile.vue [hunk 4]
```diff
@@ -166,6 +232,35 @@ export default {
...
+      localStorage.jobTheme = theme
+
+      // set job theme in app
+      const ele = document.getElementById('app')
+      for (const className of ele.classList) {
+        if (className.startsWith('job_theme--')) {
+          ele.classList.remove(className)
+        }
+      }
+      ele.classList.add(`job_theme--${theme}`)
```
[the remark this answers is not in the document]
Comment (oliver-sanders) [reply src/views/UserProfile.vue:247]: If we can move the value to Vuex then that would avoid the need for this.

File: src/App.vue [hunk 5]
```diff
@@ -16,7 +16,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 -->
 
 <template>
-  <v-app>
+  <v-app class="job_theme--default">
```
[the remark this answers is not in the document]
Comment (oliver-sanders) [reply src/App.vue]: Keen to keep this global so that it doesn't have to be explicitly enabled in each case.

**Comment by oliver-sanders** [comment]
Still got that CSS quirk to fix.

**Comment by oliver-sanders** [comment]
That should squash the CSS bug.

File: src/views/UserProfile.vue [hunk 6]
```diff
@@ -121,6 +126,50 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
                 </v-btn>
               </v-flex>
             </v-layout>
+
+            <v-layout row align-center wrap>
+              <v-flex xs3>
+                <span>Colour Theme</span>
```
[the remark this answers is not in the document]
Comment (oliver-sanders) [reply src/views/UserProfile.vue:132]: I use en_GB without really thinking about it for obvious reasons. We do seem to interchangeably use en_GB with en_US.

The main contributors language variants en_GB, en_NZ, en_AU are all pretty closely aligned.

**Review by kinow — APPROVED** [review]
Looks good to me! Tested and worked as expected. Needs only the changelog+tests :+1: 

image

**Comment by kinow** [comment]
@oliver-sanders in case @hjoliver is busy with other issues, we can assign @dwsutherland or another UK team member to review this task, doing more of a functional review.

**Comment by oliver-sanders** [comment]
I'll try and wrap my head around the tests shortly...

**Comment by oliver-sanders** [comment]
(rebased)

**Comment by oliver-sanders** [comment]
Added cypress test.

File: src/views/UserProfile.vue [hunk 7]
```diff
@@ -121,6 +126,53 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+                      <th>State</th>
+                      <th
+                        v-for="theme in jobThemes"
+                        v-bind:key="theme"
+                      >
+                        {{theme}}
+                      </th>
+                    </tr>
+                    <tr
+                    >
```
Comment (MetRonnie) [inline src/views/UserProfile.vue:146]: Niggle
```suggestion
                    <tr>
```

**Review by MetRonnie — APPROVED** [review]
I don't have much vue experience but LGTM. Tested locally and it's working

**Comment by hjoliver** [comment]
Tested here too, looks good.

**Pull request merged**